.github/workflows/deploy-docs.yml

```diff
@@ -0,0 +1,24 @@
+name: Publish docs
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish-docs:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        run: |
+          python -m venv venv
+          . ./venv/bin/activate
+          pip install -r requirements-docs.txt
+
+      - name: Deploy documentation
+        run: |
+          . ./venv/bin/activate
+          mkdocs gh-deploy --force
```

Comment:
I think we will want this to change cause I don't think we should use GitHub pages for this project.